py_zipkin/zipkin.py

```diff
@@ -285,7 +288,7 @@ def start(self):
 
```

Comment:
I think you also need to check `self.firehose_handler` in line 246 (self.perform_logging = bool(...)).

github doesn't let me add a comment there since it's far from your changes :(

Reply:
Good call, though I'm not really sure what this code path would even do. In what case would you have no sample rate, no passed in zipkin attrs, and no ongoing trace? Maybe if you called some function that is usually an "inner span"? I dunno. Anyway, I added a test for it and fixed the issue.